### ISSUE TITLE
hls output from api

### ISSUE THREAD

**Issue by roneyrogerio** [chrippa/livestreamer#1493]
it is possible to save a stream inside the API to the local disk on hls/M3U8 format?

**Comment by scottbernstein-zz** [comment]
use the -o option and it will save the file to local disk.

**Comment by roneyrogerio** [comment]
Hello, thanks for answering.
This will not help me, as I said I am using the api and not the command line, in addition to this option to save a file and I believe this is not a hls live streaming, the command line solution is to use ffmpeg .

**Comment by steven7851** [comment]
`livestreamer -O [URL] | ffmpeg -re -i - -bsf:a aac_adtstoasc -vcodec copy -c copy output.ts`
`livestreamer -p ffmpeg -a "-re -i - -bsf:a aac_adtstoasc -vcodec copy -c copy output.ts" [URL]`

If this is not you want, maybe you can try `youtube-dl` which has more options.

**Comment by roneyrogerio** [comment]
Hello Steven, thanks for the reply.
I will be more specific this time.
I've done this from the command line using ffmpeg, however want to do this using the api with python.
Documentation has this example

```
>>> import livestreamer
>>> streams = livestreamer.streams("http://twitch.tv/day9tv")
>>> stream = streams["source"]
>>> fd = stream.open()
>>> data = fd.read(1024)
>>> fd.close()
```

What I could tell is that the variable fd is just a piece of video in real time always updated, how can I continually read the fd file-like object and save him as a live hls/M3U8?
I'll look at this youtube-dl, thanks.

**Comment by intact** [comment]
You can try something like:

``` python
#!/usr/bin/python3

import livestreamer

streams = livestreamer.streams("http://twitch.tv/bobross")
stream = streams["source"]

stream_fd = stream.open()

try:
    with open("video.ts", "wb") as output_fd:
        for data in stream_fd:
            output_fd.write(data)
finally:
    stream_fd.close()
```

but some progress info will be nice. For console you can use progress from livestreamer_cli.utils:

``` python
#!/usr/bin/python3

import livestreamer
from livestreamer_cli.utils import progress

streams = livestreamer.streams("http://twitch.tv/bobross")
stream = streams["source"]

stream_fd = stream.open()
stream_fd = progress(stream_fd, "video.ts")

try:
    with open("video.ts", "wb") as output_fd:
        for data in stream_fd:
            output_fd.write(data)
finally:
    stream_fd.close()
```

**Comment by roneyrogerio** [comment]
Thank intact.
Yes, we are on the way, but I was thinking, to create a segmented hls I'll need to take the time of each segment, inevitably I will have to pass the video through a player, encoder or something, I'm wrong? Searching a bit I found this https://github.com/superdump/hlss/blob/master/hlss.py, it is possible to join it with the example gstreamer in the documentation? I could not test because my installation of gstremer giving error at this time, but it is theoretically possible?

**Comment by intact** [comment]
If you realy need save hls segments on your disk, you can use something like [hls-client](https://github.com/globocom/hlsclient)

**Comment by roneyrogerio** [comment]
No, I do not need to save to disk, the initial idea was to save on disk but then create a web server pointing to the folder with the M3U8 and .ts.
But my doubt now is whether it is possible to use the code https://github.com/superdump/hlss/blob/master/hlss.py , instead of an input file, edit the code using the sample player gstrem in livestreamer api documentation, and put a live streaming at the buffer, this is theoretically possible?

**Comment by intact** [comment]
Yes, you can use appsrc element.

Btw why do you need mpeg-ts segments?

**Comment by roneyrogerio** [comment]
the idea is to use the cache nginx, if the http flow is continuous I believe that the cache will not work
